docs/development/getting-started.rst

```diff
@@ -0,0 +1,96 @@
+===============
+Getting Started
+===============
+
+We’re pleased that you are interested in working on pip.
+
+This document is meant to get you setup to work on pip and to act as a guide and
+reference to the the development setup. If you face any issues during this
+process, please `open an issue`_ about it on the issue tracker.
+
+Development tools
+=================
+
+pip uses :pypi:`tox` for testing against multiple different Python environments
+and ensuring reproducible environments for linting and building documentation.
+
+For developing pip, you need to install ``tox`` on your system. Often, you can
+just do ``python -m pip install tox`` to install and use it.
+
+Running Tests
+-------------
+
+pip uses the :pypi:`pytest` test framework, :pypi:`mock` and :pypi:`pretend`
+for testing. These are automatically installed by tox for running the tests.
+
+To run tests locally, run:
+
+.. code-block:: console
+
+    $ tox -e py36
+
+The example above runs tests against Python 3.6. You can also use other
+versions like ``py27`` and ``pypy3``.
+
+pip's test suite requires supported version control tools (subversion, bazaar,
+git, and mercurial) to be installed. If you are missing one of the VCS tools,
+you can tell pip to skip those tests:
+
+.. code-block:: console
+
+    $ tox -e py36 -- -k "not svn"
+    $ tox -e py36 -- -k "not (svn or git)"
+
+Running Linters
+---------------
+
+pip uses :pypi:`flake8` and :pypi:`isort` for linting the codebase. These
+ensure that the codebase is in compliance with :pep:`8` and the imports are
+consistently ordered and styled.
+
+To use linters locally, run:
+
+.. code-block:: console
+
+    $ tox -e lint-py2
+    $ tox -e lint-py3
+
+The above commands run the linters on Python 2 followed by Python 3.
```

Comment:
As noted above, this doesn't work on Windows where there's no `python2` and `python3` commands. Also, the user may not have both Python 2 and Python 3 installed.

Reply:
I'd thought about the user not having Python 2/3 installed and figured that most users would realize why they can't run the linter though, when it fails saying could not find python3 or something along those lines.